src/components/dashboard/EditClient.tsx

```diff
@@ -49,11 +52,29 @@ const Profile = () => {
       onCompleted={onCompleted}
       top={STICKY_BAR_HEIGHT + CONTEXT_HEADER_HEIGHT}
       title={
-        <>
-          <Typography variant='h3' sx={{ pt: 0, pb: 4 }}>
+        <Stack direction='row' justifyContent='space-between'>
+          <Typography variant='h3' sx={{ pt: 0, pb: 4 }} flexGrow={1}>
             Edit Client Details
           </Typography>
-        </>
+          {client.id && (
+            <div>
+              <RootPermissionsFilter permissions='canDeleteClients'>
```

Comment:
Can you add `canDeleteClients` to the client access fragment so that we don't get the glitchy flashing appearance again here?

Reply:
I'm not seeing a glitchy flashing appearance, can you describe in more detail what you're seeing that looks glitchy? 

Also `canDeleteClients` is also not a client-based permission, it's a root permission, so it should be fetched on the query access fragment, not on the client access fragment.